colcon_core/environment/__init__.py

```diff
@@ -126,8 +126,12 @@ def create_environment_scripts(
     # create file containing the runtime dependencies
     path = prefix_path / get_relative_package_index_path() / pkg.name
     path.parent.mkdir(parents=True, exist_ok=True)
+    dependencies = [dependency.name
+                    for dependency in pkg.dependencies.get('run', set())]
+    # TODO: Is this supposed to be in topological order? I don't think the
+    # set is ordered?
```

Comment:
No, these dependencies are not ordered. Alphabetical only for deterministic results.

Reply:
I think the comment it still valid as it is. The file contains "the runtime dependencies" and it is being used to "source all packages in topological order", It doesn't imply that the order in the file itself is ordered.